### PR TITLE
remove references to Hugo versions

### LINKED ISSUE #68: Drop unneeded HUGO_VERSION from netlify.toml
Given that this template repo uses the [hugo-extended](https://www.npmjs.com/package/hugo-extended) npm package, IMHO we don't need to (actually, shouldn't) specifying a Hugo version in `netlify.toml`:

https://github.com/cncf/hugo-netlify-starter/blob/9ce9f2d835f216db463c4872b9ddf984eb4ad42f/netlify.toml#L5-L6

Let's keep things [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and configure the required Hugo version in only one place.

Related: #66

### PR DESCRIPTION
**Pull request by kapunahelewong**
Fixes #68 

Preview: https://deploy-preview-96--cncf-hugo-starter.netlify.app/

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -2,9 +2,6 @@
 publish = "public"
 command = "make production-build"
 
-[build.environment]
-HUGO_VERSION = "0.75.1"
```
Comment (celestehorgan) [inline netlify.toml:6]: Can I get a little context on this PR and why it's necessary? 

Not opposed to removing this, but it does seem to be causing a build failure.

File: Makefile [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 		--disableFastRender
 
 production-build:
-	hugo \
+	npx hugo \
 		--minify
 
 preview-build:
-	hugo \
+	npx hugo \
```
Comment (celestehorgan) [inline Makefile:12]: Why `npx` versus yarn or make?

File: netlify.toml (hunk 1 above)
Comment (kapunahelewong) [reply netlify.toml:6]: https://github.com/cncf/hugo-netlify-starter/issues/68 might be helpful for context, but maybe @chalin could offer more insight?

File: Makefile (hunk 2 above)
Comment (kapunahelewong) [reply Makefile:12]: cc @chalin

File: netlify.toml (hunk 1 above)
Comment (chalin) [reply netlify.toml:6]: Yes, #68 provides the context. Thought we can discuss on Wednesday if you'd like more.

File: Makefile (hunk 2 above)
Comment (chalin) [reply Makefile:12]: `npx` is what we use to access binaries fetched via an `npm` package.

File: netlify.toml (hunk 1 above)
Comment (celestehorgan) [reply netlify.toml:6]: Nope, tysm <3

**Review by celestehorgan — APPROVED** [review]
lgtm <3

File: Makefile (hunk 2 above)
Comment (chalin) [reply Makefile:12]: The broader question is whether we want to use the `hugo-extended` package at all. We should discuss that on Wednesday.

**Review by chalin — APPROVED** [review]
LGTM

**Pull request merged**